website/docs/aiml/inferentia/index.md

```diff
@@ -0,0 +1,136 @@
+
+When a Pod requires the exposed Neuron cores, the Kubernetes scheduler can create an Inferentia node to schedule the Pod to. This is the Pod that we will schedule. Note that we have a resource requirement of `aws.amazon.com/neuron`.
+
+```file
+aiml/inferentia/inference.yaml
+```
+
+### Set up a provisioner of Karpenter for launching a node which has the Inferentia chip
+
+The lab uses Karpenter to provison an Inferentia node. Karpenter can detect the pending pod which requires Neuron cores and launch an inf1 instance which has the required Neuron cores.
+
+Karpenter requires a provisioner to provision nodes. This is the Karpenter provisioner that we will create:
+
+```file
+aiml/provisioner/provisioner.yaml
+```
+
+Apply the provisioner manifest:
+
+```bash
+$ kubectl apply -k /workspace/modules/aiml/provisioner/
+```
+
+### Create a pod for inference
+
+Now we can deploy a Pod for inference:
+
+```bash
+$ kubectl apply -k /workspace/modules/aiml/inferentia/
+```
+
+Karpenter detects the pending pod which needs Neuron cores and launches an inf1 instance which has the Inferentia chip. Monitor the instance provisioning with the following command:
+
+```bash test=false
+$ kubectl logs -f -n karpenter deploy/karpenter -c controller
+
+2022-10-28T08:24:42.704Z        DEBUG   controller.provisioning.cloudprovider   Created launch template, Karpenter-eks-workshop-cluster-3507260904097783831  {"commit": "37c8653", "provisioner": "default"}
+2022-10-28T08:24:45.125Z        INFO    controller.provisioning.cloudprovider   Launched instance: i-09ddba6280017ae4d, hostname: ip-100-64-10-250.ap-northeast-1.compute.internal, type: inf1.xlarge, zone: ap-northeast-1a, capacityType: spot  {"commit": "37c8653", "provisioner": "default"}
+2022-10-28T08:24:45.136Z        INFO    controller.provisioning Created node with 1 pods requesting {"aws.amazon.com/neuron":"1","cpu":"125m","pods":"6"} from types inf1.xlarge, inf1.2xlarge, inf1.6xlarge, inf1.24xlarge       {"commit": "37c8653", "provisioner": "default"}
+2022-10-28T08:24:45.136Z        INFO    controller.provisioning Waiting for unschedulable pods  {"commit": "37c8653"}
+```
+
+The inference pod should be scheduled on the node provisioned by Karpenter. Check if the Pod is in it's ready state:
+
+```bash timeout=360
+$ kubectl -n aiml wait --for=condition=Ready --timeout=5m pod/inference
+```
+
+### Run a inference
+
+This is the code that we will be using to run inference using a Neuron core on Inferentia:
+
+```file
+aiml/inferentia/inference.py
+```
+
+Next we copy this code to the Pod, download our previously uploaded model, and run the code:
+
+```bash
+$ kubectl -n aiml cp /workspace/modules/aiml/inferentia/inference.py inference:/
```

Comment:
is this the recommended way to demonstrate training models in Kubernetes? it seems odd to do `execs` and `cps`.

Reply:
Agreed. I think we should change both of the training and inference. 
* For training I think we could add the code to the container and run that container as a job.
* For the inference we should update the code to run as an REST API server. This whey we can CURL the container either directly or through the ingress. I think that would be more inline with a real world example.

What do you think?

Reply:
Leaving as is per offline conversation.